### DISCUSSION TITLE
Don't force customer to enter shipping info if they want local pickup

### DISCUSSION THREAD

**Discussion by steph-query** [swellstores/community#452] [Help] [answered]
On the Swell hosted checkout page, the customer is forced to enter a shipping address before they choose their shipping method. I understand that where they want to have the order shipped will inform which shipping methods are available, but it's also bad UX to force a customer to enter a bunch of unnecessary information if they plan to pick the order up locally.

I sell a lot of product to people checking out on their phones at in-person events, and it's helpful to have this as smooth as possible to expedite the process. Is the only way to achieve this to build a fully custom checkout page?

**Comment by ericingram** [comment]
Do you only have a local pickup option configured in the store?

**Reply by steph-query** [reply]
I have both shipping and Local Pickup

**Reply by ericingram** [reply, marked as answer]
The shipping info is used to determine which methods are available, and local pickup is considered one of those methods currently.

We might consider making local pickup an option on the first page in the near future.

**Reply by steph-query** [reply]
Thank you! Yes I understand that addresses determine the shipping options. It would just be great from a UX perspective to put Local pickup on the first page so a user doesn't have to enter that info if they won't end up needing it.
Cheers!
